src/main/java/org/betonquest/betonquest/objectives/BlockObjective.java

```diff
@@ -78,4 +127,35 @@ public void stop() {
         HandlerList.unregisterAll(this);
     }
 
+    private boolean checkLocation(final Location loc, final Profile profile) {
+        try {
+            if (location != null) {
+                if (region != null) {
+                    return isInRange(loc, profile);
+                }
+                return loc.getBlock().getLocation().equals(location.getLocation(profile));
+            }
+        } catch (QuestRuntimeException e) {
+            logger.error(instruction.getPackage(), e.getMessage());
+        }
+        return false;
+    }
```

Comment:
This will return `false` if both location arguments are `null`, thus entirely blocking `block` objectives that don't set them.

```suggestion
    private boolean checkLocation(final Location loc, final Profile profile) {
        try {
            if (location == null) {
                return true;
            }
            if (region == null) {
                return loc.getBlock().getLocation().equals(location.getLocation(profile));
            }
            return isInRange(loc, profile);
        } catch (QuestRuntimeException e) {
            logger.error(instruction.getPackage(), e.getMessage());
            return false;
        }
    }
```